app/services/resource_compare_service.rb

```diff
@@ -32,52 +32,49 @@ def identical?
   private
 
     def compare_resources
-      [:titles, :creators, :contributors, :related_objects].each do |field|
-        compare_objects(field)
-      end
-      [:description, :publisher, :publication_year,
-       :resource_type, :resource_type_general,
-       :doi, :ark, :version_number].each do |field|
-        compare_simple_values(field)
-      end
-      compare_rights
-      [:keywords, :collection_tags].each do |field|
-        compare_arrays(field)
+      @before.as_json.keys.map(&:to_sym).each do |method_sym|
+        before_value = @before.send(method_sym)
+        after_value = @after.send(method_sym)
+        if before_value.is_a?(Array)
+          next if before_value.empty? && after_value.empty?
```

Comment:
I wonder if you could use equality to exit early for all cases and simplify the code a bit.
Aka on line 37 you could put `next if before_value.to_json == after_value.to_json`
I did a bit of fiddling in the console and nil.to_json is equivalent to nil.to_json and also the full object equivalence will work with a to_json.
```
[14] pry(main)> before = PDCMetadata::Resource.new(doi:"abc123")
=> #<PDCMetadata::Resource:0x00000001155940a0
 @ark=nil,
 @collection_tags=[],
 @contributors=[],
 @creators=[],
 @description=nil,
 @doi="abc123",
 @funders=[],
 @keywords=[],
 @publication_year=2023,
 @publisher="Princeton University",
 @related_objects=[],
 @resource_type="Dataset",
 @resource_type_general="Dataset",
 @rights=nil,
 @titles=[],
 @version_number="1">
[15] pry(main)> after = PDCMetadata::Resource.new(doi:"abc123")
=> #<PDCMetadata::Resource:0x0000000114e15630
 @ark=nil,
 @collection_tags=[],
 @contributors=[],
 @creators=[],
 @description=nil,
 @doi="abc123",
 @funders=[],
 @keywords=[],
 @publication_year=2023,
 @publisher="Princeton University",
 @related_objects=[],
 @resource_type="Dataset",
 @resource_type_general="Dataset",
 @rights=nil,
 @titles=[],
 @version_number="1">
[16] pry(main)> before == after
=> false
[17] pry(main)> before.to_json == after.to_json
=> true
[18] pry(main)> before = nil
=> nil
[19] pry(main)> after = nil
=> nil
[20] pry(main)> before.to_json == after.to_json
```

Reply:
Sure! Let me merge the other suggestions, and then I'll add that locally.